### PR TITLE
Implement SubmitProposalModal

### PR DESCRIPTION
**Pull request by hochiw**
## Features
- Implemented govAPI
- Implemented SubmitProposalForm and Modal
- Safe guard disconnected user and prompt them to connect to wallet if not done so before submitting proposal
- Extracted queryClient into a standalone provider
- Temporary fetches minimal deposit param directly from chain, will handle when working on GovParamPanel

## Screenshots
| Desktop | Mobile |
|---------|--------|
|Screenshot 2022-06-02 at 5 56 53 PM|Screenshot 2022-06-02 at 5 56 59 PM|

based on #82 
refs #66

### CODE REVIEW

**Comment by chihimng** [comment]
plz rebase 🙏

**Comment by hochiw** [comment]
rebased

File: react-app/src/components/CreateProposalScreen/CreateProposalScreen.tsx [hunk 1]
```diff
@@ -1,14 +1,100 @@
...
+  );
+
+  const onSubmitSubmitProposal = useCallback(
+    (values: SubmitProposalFormValues) => {
+      if (wallet.status !== ConnectionStatus.Connected) return;
+      const deposit = new BigNumber(values.amount).shiftedBy(
+        chainInfo.currency.coinDecimals
+      );
+
+      govAPI
```
Comment (chihimng) [inline react-app/src/components/CreateProposalScreen/CreateProposalScreen.tsx:59]: can we use async await here?

**Review by chihimng — CHANGES_REQUESTED** [review]
also, we might want to introduce a util to display any caught errors?

File: react-app/src/components/CreateProposalScreen/CreateProposalScreen.tsx (hunk 1 above)
Comment (hochiw) [reply react-app/src/components/CreateProposalScreen/CreateProposalScreen.tsx:59]: in this case we might want to align all the previous calls for the shortcuts

Comment (hochiw) [reply react-app/src/components/CreateProposalScreen/CreateProposalScreen.tsx:59]: no time, I will open issue for the alignment

**Comment by hochiw** [comment]
> also, we might want to introduce a util to display any caught errors?

the toast already reflects error state, do you want it to show the exact error message as well?

**Comment by chihimng** [comment]
> the toast already reflects error state, do you want it to show the exact error message as well?

nvm then

**Pull request merged**